src/Microsoft.AspNetCore.Mvc.Razor.Extensions/InstrumentationPass.cs

```diff
@@ -45,33 +42,33 @@ private static void AddInstrumentation(InstrumentationItem item)
                         item.IsLiteral ? "true" : "false")
                 });
 
-            var endNode = new CSharpCodeIRNode()
-            {
-                Parent = item.Node.Parent
-            };
+            var endNode = new CSharpCodeIRNode();
             RazorIRBuilder.Create(endNode)
                 .Add(new RazorIRToken()
                 {
                     Kind = RazorIRToken.TokenKind.CSharp,
                     Content = string.Format("{0}();", endContextMethodName)
                 });
 
-            var nodeIndex = item.Node.Parent.Children.IndexOf(item.Node);
-            item.Node.Parent.Children.Insert(nodeIndex, beginNode);
-            item.Node.Parent.Children.Insert(nodeIndex + 2, endNode);
+            var nodeIndex = item.Parent.Children.IndexOf(item.Node);
+            item.Parent.Children.Insert(nodeIndex, beginNode);
+            item.Parent.Children.Insert(nodeIndex + 2, endNode);
         }
 
         private struct InstrumentationItem
         {
-            public InstrumentationItem(RazorIRNode node, bool isLiteral, SourceSpan source)
+            public InstrumentationItem(RazorIRNode node, RazorIRNode parent, bool isLiteral, SourceSpan source)
```

Comment:
This is the new basic pattern for rewriters - grab the parent and the node during the visiting phase, and then act on it later.